management/src/clusterm/manager/api.go

```diff
@@ -141,36 +137,35 @@ func validateAndSanitizeEmptyExtraVars(errorPrefix, extraVars string) (string, e
 	return extraVars, nil
 }
 
-func (m *Manager) nodesCommission(tags []string, sanitizedExtraVars string) error {
-	me := newWaitableEvent(newCommissionEvent(m, tags, sanitizedExtraVars))
+func (m *Manager) nodesCommission(req *APIRequest, extraVars string) error {
+	if !IsValidHostGroup(req.HostGroup) {
```

Comment:
We should move this validation to the event handler itself, where rest of the request validations occur. This will keep all the validation logic close together making it easy to maintain.